testlapack/general.go

```diff
@@ -1016,16 +1016,12 @@ func isUpperTriangular(a blas64.General) bool {
 
 // randomSparseGeneral returns an m×n general matrix whose each element is set
 // to a non-zero value with probability p.
-func randomSparseGeneral(m, n, stride int, p float64, rnd *rand.Rand) blas64.General {
-	a := nanGeneral(m, n, stride)
-	for i := 0; i < m; i++ {
-		for j := 0; j < n; j++ {
-			if rnd.NormFloat64() < p {
-				a.Data[i*stride+j] = rnd.NormFloat64()
-			} else {
-				a.Data[i*stride+j] = 0
-			}
-		}
+func randomSparseGeneral(m, n, stride int, nonzeros int, rnd *rand.Rand) blas64.General {
+	a := zeros(m, n, stride)
+	for k := 0; k < nonzeros; k++ {
+		i := rnd.Intn(n)
+		j := rnd.Intn(n)
+		a.Data[i*stride+j] = float64(j+1) * rnd.NormFloat64()
```

Comment:
One doesn't get away with anything here :-) Renamed to `unbalancedSparseGeneral`.

Now that I specify the number of nonzeros the unbalancing is not strictly necessary, the row and column norms differ enough just by absence or presence of zeros. I help it a bit by multiplying with row or column index randomly.